src/Canvas/canvasFunctions.js

```diff
@@ -79,88 +65,107 @@ function drawWeight(ctx, points, weight) {
   ctx.fillStyle = "#111";
   ctx.closePath();
 }
-export function drawEdge(ctx, node1, node2, size, color = null) {
+
+/*
+  let x, y define the position of the first node of edge.
+  let x1, y1 define the position of the second node of edge.
+  to draw an edge we need to move these position to the edge(corner) of the both nodes.
+  this function return the new cordinates as follow :
+
+  let's move the position of each node  to the edge(corner)  but we have two cases, moving the new position to the left or the right direction of the original position
+  but as we don't know which direction we should move, we calcluate the distance from the new position to the other node , for example the if  left direction increase the
+  distance between the two original point we draw the left direction instead and vise versa
+
+  we will do the same action for the both node
+ */
+export function getCorrectPoints(x, y, x1, y1, size) {
+  let x_move, y_move, x1_move, y1_move;
+  // first node
+  let p = tranlsate_point(x1, y1, calcSlope(x, y, x1, y1), size, 1);
+  let p2 = tranlsate_point(x1, y1, calcSlope(x, y, x1, y1), size, -1);
+  [x1_move, y1_move] = getDist(x, y, p[0], p[1]) < getDist(x, y, x1, y1) ? p : p2;
+  /************/
+  // second node
+  p = tranlsate_point(x, y, calcSlope(x, y, x1, y1), size, 1);
+  p2 = tranlsate_point(x, y, calcSlope(x, y, x1, y1), size, -1);
+  [x_move, y_move] = getDist(x1, y1, p[0], p[1]) < getDist(x, y, x1, y1) ? p : p2;
+
+  return [x_move, y_move, x1_move, y1_move];
+};
+
+export async function drawEdge(ctx, node1, node2, size, color = null) {
```

Comment:
why does drawEdge need to be an async function?!

Reply:
it was kind of testing for the async method and I forgot to delete it after that.
sorry for the inconvenience.